### DISCUSSION TITLE
Questions about sessions

### DISCUSSION THREAD

**Discussion by tom-i** [qutebrowser/qutebrowser#7080] [General]
1. opening browser with `session_name1`, then open `session-load session_name2` then closing window with `session_name1` and then `session_name2` made, that URLs from `session_name1` will be overwritten by `session_name2` am I understanding correctly?
2. So If I want to use window with new session I should start new fresh qutebrowser, so this new window will be with new session_name, or better question, is it possible, to open new qutebrowser with specific sesssion name as parameter? Because `qutebrowser --restore session_name` didn't open URLs from that session if I have openned many sessions in other windows
3. Is it possible to check which session am I using on specific window?

**Comment by The-Compiler** [comment]
1. Correct, closing the last window saves to the last saved session. Either load with `:session-load --temp` or set `session.default_name` to prevent that.
2. All open qutebrowser windows belong to the same instances, and sessions are not bound to windows.
3. See above - there is no such concepts as "using a session in a specific window". Also see #572.

**Reply by tom-i** [reply]
Aha, I see that session per window is not implemented. It's sad :-(
Mainly becuase I'm using qutebrowser for my personal and work stuff.. so I have a lot of sessions for each customer and so on.. :-/

**Reply by ssantana92** [reply]
This is old, but for anyone ending up here like I did: If you constrain your sessions to a single window you can use `:w -o` or `:session-save -o` to save only  the currently focused window to a session file. This is what I do, but you need to mentally keep track of which window belongs to which session. I have also set session.default_name=default and periodically save a backup session of it (default.bak) in case I accidentally overwrite the default session.
